src/utils/ethers-util.ts

```diff
@@ -51,89 +45,52 @@ const handleErrorCode = (code: number, message?: string): string => {
-  return Promise.race([promise, new Promise((resolve, reject) => setTimeout(reject, timeout))]);
-}
-
-async function connectEvmProvider(provider: Provider, chains: ChainsProps): Promise<string> {
-  switch (provider) {
-    case Provider.WalletConnect:
-      return await useWalletConnectProvider(chains);
-    default:
-      return await useExtensionProvider(provider);
-  }
-}
-
-function disconnectEvmProvider(provider?: Nullable<Provider>): void {
-  ethereumProvider?.disconnect?.();
-  // don't wait promise execution, that's for wallets lifecycle
-  revokeWalletAccounts();
-}
-
-function createWeb3Instance(provider: any) {
-  ethereumProvider = provider;
-  // 'any' - because ethers throws errors after network switch
-  ethersInstance = new ethers.BrowserProvider(ethereumProvider, 'any');
-}
-
-async function useExtensionProvider(provider: Provider): Promise<string> {
-  const injectedWindow = window as any;
-
-  let ethereumProvider!: any;
-
-  switch (provider) {
-    case Provider.Metamask:
-      ethereumProvider = await detectEthereumProvider({ mustBeMetaMask: true, timeout: 0 });
-      break;
-    case Provider.SubWallet:
-      ethereumProvider = injectedWindow.SubWallet;
-      break;
-    case Provider.TrustWallet:
-      ethereumProvider = injectedWindow.trustwallet;
-      break;
-    // case Provider.Fearless:
-    //   ethereumProvider = injectedWindow.fearlessWallet;
-    //   break;
-    default:
-      throw new Error('Unknown provider');
-  }
-
-  if (!ethereumProvider) throw new Error(installExtensionKey);
-
-  createWeb3Instance(ethereumProvider);
-
-  const accounts = await requestWalletAccounts();
+export const handleRpcProviderError = (error: any): string => {
+  const { code, message } = getErrorCodeMessage(error);
 
-  return accounts[0];
-}
+  return handleErrorCode(code, message);
+};
 
-async function useWalletConnectProvider(chainProps: ChainsProps): Promise<string> {
+async function connectEvmProvider(appEvmProvider: AppEIPProvider, chainsProps: ChainsProps): Promise<string> {
   try {
-    const ethereumProvider = await getWcEthereumProvider(chainProps);
+    const ethereumProvider = await appEvmProvider.getProvider(chainsProps);
+
+    if (!ethereumProvider) throw new Error(installExtensionKey);
 
-    await ethereumProvider.connect();
+    await ethereumProvider.connect?.();
 
     createWeb3Instance(ethereumProvider);
 
-    return await getAccount();
+    switch (appEvmProvider.uuid) {
+      case PredefinedProvider.WalletConnect: {
+        return await getAccount();
+      }
+      default: {
+        const accounts = await requestWalletAccounts();
+        return accounts[0];
+      }
+    }
   } catch (error: any) {
+    // user rejected request
+    if (error.code === PROVIDER_ERROR.UserRejectedRequest) {
+      return '';
+    }
     // user cancelled qr modal
     if (error.message === 'Connection request reset. Please try again.') {
       return '';
```

Comment:
Shouldn't we handle other errors?

Reply:
No, this WalletConnect error handlers, only 2 correct cases, then user closes modal, or reject approval in his wallet